### ISSUE TITLE
Allocating an array larger than 2MBs in a 'local' block attempts communication 

### ISSUE THREAD

**Issue by LouisJenkinsCS** [chapel-lang/chapel#10536]
Due to the heuristic described below for whether initialization should be performed in parallel, there is a GET to locale 0.

https://github.com/chapel-lang/chapel/blob/b0d4683338f26fbc2ad410071e5cf7325d64b73a/modules/internal/ChapelBase.chpl#L727-L744

The actual algorithm can be seen slightly below it, where it decides to perform a `parallelInit`.

https://github.com/chapel-lang/chapel/blob/b0d4683338f26fbc2ad410071e5cf7325d64b73a/modules/internal/ChapelBase.chpl#L748-L759

The culprit is this line here, where it will attempt to access `DummyLocale`

https://github.com/chapel-lang/chapel/blob/b0d4683338f26fbc2ad410071e5cf7325d64b73a/modules/internal/ChapelBase.chpl#L765

Which is allocated on locale 0...

https://github.com/chapel-lang/chapel/blob/b0d4683338f26fbc2ad410071e5cf7325d64b73a/modules/internal/ChapelLocale.chpl#L582

**MWE**

```chpl
 on Locales[1] {                                                                                                                                                                                                                                                                
   local {                                                                                                                                                                                                                                                                      
     var arr : [1..2 * 1024 * 1024 + 1] int;                                                                                                                                                                                                                                    
     assert(true); // Don't compile away...                                                                                                                                                                                                                                     
   }                                                                                                                                                                                                                                                                            
 } 
```

**Output**

```
$CHPL_HOME/modules/internal/ChapelLocale.chpl:582: error: cannot access remote data in local block
```

**Comment by LouisJenkinsCS** [comment]
For a quick-fix, would it be sufficient to compare the memory address? As in...

```chpl
   const dummyAddr = __primitive("cast", uint, __primitive("_wide_get_addr", dummyLocale));
   const ourAddr = __primitive("cast", uint, here);
   if ourAddr == dummyAddr { 
     initMethod = ArrayInit.serialInit; 
   } 
```

**Comment by ronawho** [comment]
Does replacing the `if here == dummyLocale` with `if rootLocaleInitialized` resolve the issue?

**Comment by ronawho** [comment]
Whoops, got that condition wrong -- should be `if !rootLocaleInitialized`

**Comment by ronawho** [comment]
Merged to master -- I'll let you verify and check it off of the mega issue

**Comment by LouisJenkinsCS** [comment]
Hm, I'm not sure if I'm hitting another bug besides this, but its also related to 'accessing remote data in local blocks' in `Cyclic`

https://github.com/chapel-lang/chapel/blob/b0d4683338f26fbc2ad410071e5cf7325d64b73a/modules/dists/CyclicDist.chpl#L777-L780

That's another issue entirely though and it does solve this issue so I'll mark it off.

**Comment by LouisJenkinsCS** [comment]
Also thank you @ronawho for helping with this so quickly.
